### PR TITLE
feat: 添加doc: JavaScript的工作原理: 密码学 + 如何应对中间人攻击

### PR DESCRIPTION
**Pull request by zheng03**
task basic 44 文章 解读

### CODE REVIEW

**Review by jhfnetboy — APPROVED** [review]
I will check later

File: basic/44-javascript-cryptography/readme.md [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+密码学意指保护信息和通信的过程，以便只有发送方和预期的接收方可以访问它们。密码学使用了很多技术来确保护通信过程的安全。这些技术包括基于密码的加密和解密、使用各种算法对通信过程进行Hash计算以及签名生成和验证。
+
+由于很多移动应用在互联网上通信中使用了 JavaScript 相关技术, 因此需要了解在 JavaScript 中密码学是如何工作的。接下来的章节，我们将讨论下 JavaScript 的 Web Cryptography API 以及它如何支持加密。
+
+## JavaScript 的 Web 加密 API
+由于确保通过 Internet 进行安全通信很重要，因此一些 Web 浏览器已经实现了[crypto](https://developer.mozilla.org/en-US/docs/Web/API/Crypto)接口。但是，此接口定义不明确或密码学上不健全。JavaScript 的 Web 加密 API 提供了一个定义良好的接口，称为[SubtleCrypto](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto)
+
+JavaScript 的网络加密 API 允许开发人员将基本的加密功能合并到他们的应用程序中，而无需第三方库。您可以签署文档、执行身份验证并执行通信的整体完整性检查。
+
+例如，您可以通过运行以下代码来获取 8 位无符号整数数组的加密安全随机数据：
```
Comment (luckyyang) [inline basic/44-javascript-cryptography/readme.md:19]: Better to use code block format for code example such as:

```javascript
    const secure = window.crypto.getRandomValues(new Uint8Array(10));
    console.log(secure);
```

**Review by jhfnetboy — APPROVED** [review]
great job, do we have any demo to show the practice ?

**Review by maroonstar — APPROVED** [review]
Great，please read the link https://github.com/rebase-network/Dapp-Learning/blob/main/docs/dao.md
add WeChat to join our developer group

**Comment by zheng03** [comment]
> great job, do we have any demo to show the practice ?

yes, we do , will post later in another pr.

**Pull request merged**